src/filenames/bcl_processor.rs

```diff
@@ -174,69 +189,43 @@ pub fn group_samples(
 pub fn find_flowcell_fastqs(
     path: impl AsRef<Path>,
 ) -> Result<Vec<(BclProcessorFileGroup, InputFastqs)>> {
-    let demux_files = get_demux_files(path.as_ref())?
+    Ok(get_demux_files(path.as_ref())?
         .into_iter()
         .sorted()
-        .group_by(|(info, _)| info.group.clone());
-    let flowcell_fastqs = demux_files.into_iter().map(|(group, files)| {
-        let my_files: Vec<_> = files.collect();
-
-        let ra = my_files
-            .clone()
-            .into_iter()
-            .find(|(info, _)| info.read == "RA")
-            .expect("couldn't find RA read");
-        let i1 = my_files
-            .clone()
-            .into_iter()
-            .find(|(info, _)| info.read == "I1");
-        let i2 = my_files.into_iter().find(|(info, _)| info.read == "I2");
-
-        let fastqs = crate::read_pair_iter::InputFastqs {
-            r1: ra.1.to_str().unwrap().to_string(),
-            r2: None,
-            i1: i1.map(|x| x.1.to_str().unwrap().to_string()),
-            i2: i2.map(|x| x.1.to_str().unwrap().to_string()),
-            r1_interleaved: true,
-        };
-        (group, fastqs)
-    });
-    Ok(flowcell_fastqs.collect())
+        .group_by(|info| info.group.clone())
+        .into_iter()
+        .filter_map(|(group, group_files)| {
+            let group_files: Vec<_> = group_files.collect();
+
+            group_files
+                .iter()
+                // ignore groups that don't have an RA file
```

Comment:
I don't have the context to understand why we want to allow this condition to pass silently; would it make sense to make this behavior configurable, allowing the caller to choose between ignoring this condition or returning an error?